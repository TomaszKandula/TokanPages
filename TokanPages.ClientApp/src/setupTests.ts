// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
