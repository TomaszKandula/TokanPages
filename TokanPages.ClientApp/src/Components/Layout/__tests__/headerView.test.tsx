import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import HeaderView from "../headerView";

describe("Test component: headerView.", () => 
{
    it("Renders correctly '<HeaderView />' when content is loaded.", () => 
    {
        const tree = shallow(<HeaderView isLoading={false} content=
        {{
            photo: "",
            caption: "Welcome to my web page",
            description: "I do programming for a living...",
            action: 
            {
                text: "Read the story",
                href: "/action-link"
            }
        }}/>);
        expect(tree).toMatchSnapshot();
    });
});
