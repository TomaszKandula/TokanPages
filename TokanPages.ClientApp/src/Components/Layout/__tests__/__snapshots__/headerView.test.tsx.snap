// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component: headerView. Renders correctly '<HeaderView />' when content is loaded. 1`] = `
<section>
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-top_margin-1"
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={6}
        xs={12}
      >
        <Styled(MuiBox)
          className="makeStyles-image_box-7"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={6}
        xs={12}
      >
        <Styled(MuiBox)
          className="makeStyles-content_box-4"
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
            variant="h3"
          >
            Welcome to my web page
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-content_description-5"
            variant="h6"
          >
            I do programming for a living...
          </WithStyles(ForwardRef(Typography))>
          <Styled(MuiBox)
            mt={4}
          >
            <ActiveButton />
          </Styled(MuiBox)>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</section>
`;
